Tools/cases_generator/analyzer.py

```diff
@@ -1215,6 +1217,71 @@ def analyze_forest(forest: list[parser.AstNode]) -> Analysis:
         instructions, uops, families, pseudos, labels, opmap, first_arg, min_instrumented
     )
 
+#Simple heuristic for size to avoid too much stencil duplication
+def is_large(uop: Uop) -> bool:
+    return len(list(uop.body.tokens())) > 80
+
+def get_uop_cache_depths(uop: Uop) -> Iterator[tuple[int, int, int]]:
```

Comment:
I don't like using generators here. There's no benefit as it's not like the uop sequence is huge or something. The code might be clearer without the yields and with just a list append instead. This is an optional review though.

Reply:
I think using a generator here is fine.